vllm-proxy/src/app/api/v1/openai.py

```diff
@@ -160,18 +199,49 @@ async def chat_completions(request: Request):
     # Get the JSON body from the incoming request
     request_body = await request.body()
     request_json = json.loads(request_body)
+    request_json = strip_empty_tool_calls(request_json)
+
+    # Check if the request is for streaming or non-streaming
+    is_stream = request_json.get(
+        "stream", False
+    )  # Default to streaming if not specified
+
+    modified_request_body = json.dumps(request_json).encode("utf-8")
+    if is_stream:
+        # Create a streaming response
+        return await stream_vllm_response(VLLM_URL, request_body, modified_request_body)
+    else:
+        # Handle non-streaming response
+        response_data = await non_stream_vllm_response(
+            VLLM_URL, request_body, modified_request_body
+        )
+        return JSONResponse(content=response_data)
+
+
+# VLLM completions
+@router.post("/completions", dependencies=[Depends(verify_authorization_header)])
+async def completions(request: Request):
+    # Get the JSON body from the incoming request
+    request_body = await request.body()
+    request_json = json.loads(request_body)
+    request_json = strip_empty_tool_calls(request_json)
 
     # Check if the request is for streaming or non-streaming
     is_stream = request_json.get(
-        "stream", True
+        "stream", False
     )  # Default to streaming if not specified
```

Comment:
default to non-straming

Reply:
The field of `stream` was intentionally changed from **True** to **False** because **True** would give upstream applications an unexpected result, so it was changed to False. For example, in redpill.